contrib/epee/include/net/http_server_impl_base.h

```diff
@@ -60,8 +61,16 @@ namespace epee
       const std::string& bind_ipv6_address = "::", bool use_ipv6 = false, bool require_ipv4 = true,
       std::vector<std::string> access_control_origins = std::vector<std::string>(),
       boost::optional<net_utils::http::login> user = boost::none,
-      net_utils::ssl_options_t ssl_options = net_utils::ssl_support_t::e_ssl_support_autodetect)
+      net_utils::ssl_options_t ssl_options = net_utils::ssl_support_t::e_ssl_support_autodetect,
+      const std::size_t max_public_ip_connections = 3,
+      const std::size_t max_private_ip_connections = 25,
+      const std::size_t max_connections = 100,
+      const std::size_t response_soft_limit = 25 * 1024 * 1024)
     {
+      if (max_connections < max_public_ip_connections)
+        throw std::logic_error{"Max public IP connections cannot be more than max connections"};
+      if (max_connections < max_private_ip_connections)
+        throw std::logic_error{"Max private IP connections cannot be more than max connections"};
```

Comment:
A bit pedantic, but `std::invalid_argument` would be a better fit here. It's not really a "logic error" if a higher level caller provides arguments that fail a basic test, it's just bad arguments

Reply:
Yup, forgot about that somehow.

Reply:
Done.